ucubot/Controllers/LessonSignalEndpointController.cs

```diff
@@ -24,33 +25,136 @@ public LessonSignalEndpointController(IConfiguration configuration)
+                                {
+                                    Id = Convert.ToInt32(row["Id"]),
+                                    Timestamp = (DateTime) row["Timestamp"],
+                                    Type = (LessonSignalType) Convert.ToInt32(row["SignalType"]),
+                                    UserId = row["UserId"].ToString()
+                                }
+                            );
+                        }
+
+                        return signals;
+                    }
+                }
+            }
         }
-        
+
         [HttpGet("{id}")]
         public LessonSignalDto ShowSignal(long id)
         {
             // TODO: add query to get a signal by the given id
-            return null;
+            using (MySqlConnection connection = new MySqlConnection())
+            {
+                connection.ConnectionString = _configuration.GetConnectionString("BotDatabase");
+                connection.Open();
+                MySqlCommand command = new MySqlCommand(String.Format("SELECT * FROM lesson_signal WHERE id={0}", id),
+                    connection);
+                using (MySqlDataAdapter adapter = new MySqlDataAdapter(command))
+                {
+                    using (DataTable table = new DataTable())
+                    {
+                        adapter.Fill(table);
+                        try
+                        {
+                            DataRow row = table.Rows[0];
+                            return new LessonSignalDto()
+                            {
+                                Id = Convert.ToInt32(row["Id"]),
+                                Timestamp = (DateTime) row["Timestamp"],
+                                Type = (LessonSignalType) Convert.ToInt32(row["SignalType"]),
+                                UserId = row["UserId"].ToString()
+                            };
+                        }
+                        catch (IndexOutOfRangeException e)
+                        {
+                            Console.Out.WriteLine(e.Message);
+                            return null;
+                        }
+
+
+                    }
+                }
+            }
         }
-        
+
         [HttpPost]
         public async Task<IActionResult> CreateSignal(SlackMessage message)
         {
             var userId = message.user_id;
-            var signalType = message.text.ConvertSlackMessageToSignalType();
+            try
+            {
+                var signalType = message.text.ConvertSlackMessageToSignalType();
+                using (MySqlConnection connection = new MySqlConnection())
+                {
+                    connection.ConnectionString = _configuration.GetConnectionString("BotDatabase");
+                    connection.Open();
+                    MySqlCommand command = new MySqlCommand(String.Format("INSERT INTO lesson_signal(SignalType, UserId) VALUES({0}, {1})", (int)signalType, userId), connection);
```

Comment:
fail: Microsoft.AspNetCore.Server.Kestrel[13]
app_1   |       Connection id "0HLC5452VDNE1", Request id "0HLC5452VDNE1:00000002": An unhandled exception was thrown by the application.
app_1   | MySql.Data.MySqlClient.MySqlException (0x80004005): Unknown column 'U111' in 'field list'
app_1   |    at MySql.Data.MySqlClient.MySqlStream.ReadPacket()
app_1   |    at MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
app_1   |    at MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
app_1   |    at MySql.Data.MySqlClient.MySqlDataReader.NextResult()
app_1   |    at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
app_1   |    at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
app_1   |    at System.Data.Common.DbCommand.ExecuteNonQueryAsync(CancellationToken cancellationToken)
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
app_1   |    at ucubot.Controllers.LessonSignalEndpointController.<CreateSignal>d__4.MoveNext() in /app/Controllers/LessonSignalEndpointController.cs:line 111
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeActionMethodAsync>d__12.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeNextActionFilterAsync>d__10.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeInnerFilterAsync>d__14.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeNextResourceFilter>d__22.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeFilterPipelineAsync>d__17.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeAsync>d__15.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at ucubot.Infrastructure.AspNetCoreExtensions.RequestScopingStartupFilter.<<ConfigureRequestScoping>b__3_0>d.MoveNext() in /app/Infrastructure/AspNetCoreExtensions.cs:line 94
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Hosting.Internal.RequestServicesContainerMiddleware.<Invoke>d__3.MoveNext()
app_1   | --- End of stack trace from previous location where exception was thrown ---
app_1   |    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
app_1   |    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
app_1   |    at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Frame`1.<ProcessRequestsAsync>d__2.MoveNext()